cmake/public/protobuf.cmake

```diff
@@ -0,0 +1,146 @@
+# ---[ Protobuf
+
+# We will try to use the config mode first, and then manual find.
+find_package(Protobuf CONFIG QUIET)
+if (NOT Protobuf_FOUND)
+  find_package(Protobuf MODULE QUIET)
+endif()
+
+if (TARGET protobuf::libprotobuf)
+  # Hooray. This is the most ideal situation, meaning that you either have a
+  # Protobuf config file installed (like on Windows), or you are using a
+  # modern CMake that ships with a FindProtobuf.cmake file that produces
+  # modern targets.
+  message(STATUS "Caffe2: Found protobuf with new-style protobuf targets.")
+elseif(Protobuf_FOUND OR PROTOBUF_FOUND)
+  # If the modern targets are not present, we will generate them for you for
+  # backward compatibility. This is backported from CMake's new FindProtobuf.cmake
+  # content.
+  message(STATUS "Caffe2: Found protobuf with old-style protobuf targets.")
+  if(PROTOBUF_LIBRARY)
+      if(NOT TARGET protobuf::libprotobuf)
```

Comment:
Will this question always be true?

Reply:
I think this should always be true. Let me just remove it, indeed.